resources/metadata/whitesource.yaml

```diff
@@ -0,0 +1,523 @@
+      description: List of WhiteSource projects to be included in the assessment part of the step, usually determined automatically.
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+    - name: vulnerabilityReportTitle
+      type: string
+      description: Title of vulnerability report written during the assessment phase.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: WhiteSource Security Vulnerability Report
+    - name: installCommand
+      type: string
+      description: Install command that can be used to populate the default docker image for some scenarios.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+    - name: scanType
+      type: string
+      description: Type of development stack used to implement the solution.
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+    - name: cvssSeverityLimit
+      type: string
+      description: Limit of tollerable CVSS v3 score upon assessment and in consequence fails the build, defaults to  `-1`.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: 7
+    - name: whitesource/productToken
+      type: string
+      description: Token of the WhiteSource product to be created and used for results aggregation, usually determined automatically.
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+    - name: agentParameters
+      type: string
+      description: Additional parameters passed to the Unified Agent command line.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: ''
+    secrets:
+    - name: whitesource/userTokenCredentialsId
+      type: jenkins
+    - name: whitesource/orgAdminUserTokenCredentialsId
+      type: jenkins
+    resources:
+    - name: buildDescriptor
+      type: stash
+      conditions:
+      - conditionRef: strings-equal
+        params:
+        - name: scanType
+          value: golang
+    - name: opensourceConfiguration
+      type: stash
+      conditions:
+      - conditionRef: strings-equal
+        params:
+        - name: scanType
+          value: golang
+    - name: checkmarx
+      type: stash
+      conditions:
+      - conditionRef: strings-equal
+        params:
+        - name: scanType
+          value: golang
+    - name: buildDescriptor
+      type: stash
+      conditions:
+      - conditionRef: strings-equal
+        params:
+        - name: scanType
+          value: maven
+    - name: opensourceConfiguration
+      type: stash
+      conditions:
+      - conditionRef: strings-equal
+        params:
+        - name: scanType
+          value: maven
+    - name: buildDescriptor
```

Comment:
I think the `buildDescriptor` stash is always needed and doesn't need a condition.

Reply:
Could you clarify what you mean by this? Still very new to this whole yaml step file..